tests/main/systemd-success-exit-status/task.yaml

```diff
@@ -0,0 +1,87 @@
+summary: Check that services with success-exit-status have SuccessExitStatus directive
+
+details: |
+    Verifies that a snap service with SuccessExitStatus=42 250 is considered active or successfully terminated by systemd when it exits with status 42 or 250, and that other unexpected signals do not have this effect.
+
+environment:
+    SERVICE_NAME: snap.test-snapd-success-exit-status.test-snapd-success-exit-status.service
+
+restore: |
+    snap remove --purge test-snapd-success-exit-status || true
+
+execute: |
+    echo "Given a service snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-success-exit-status
+
+    echo "When the service state is reported as active"
+    while ! systemctl show -p ActiveState "$SERVICE_NAME" | grep -Pq "ActiveState=active"; do sleep 0.5; done
+
+    echo "The service contains the SuccessExitStatus= directive"
+    systemctl cat "$SERVICE_NAME" | grep -q "SuccessExitStatus=42 250"
+
+    echo "Then systemctl reports the status of the service as loaded, active and running"
+    expected='(?s).*?Loaded: loaded .*?Active: active \(running\)'
+    systemctl status "$SERVICE_NAME" | grep -Pqz "$expected"
+
+    echo "When the service initialization is completed"
+    retry -n 20 --wait 1 stat /var/snap/test-snapd-success-exit-status/common/ready
+
+    echo "Terminate the service with success exit status 42"
+    systemctl kill -s USR1 --wait "$SERVICE_NAME"
+
+    echo "When the signal is propagated to the service"
+    retry -n 20 --wait 1 sh -c "systemctl show -p ActiveState '$SERVICE_NAME' | MATCH 'ActiveState=inactive'"
+
+    echo "Then systemctl reports the status of the service as terminated with the custom status"
+    expected='(?s).*?\(code=exited, status=42\)'
+    systemctl status "$SERVICE_NAME" | grep -Pqz "$expected"
+
+    # ==========
+
+    echo "Restart the service for another test"
+    systemctl restart "$SERVICE_NAME"
+
+    echo "When the service state is reported as active"
+    while ! systemctl show -p ActiveState "$SERVICE_NAME" | grep -Pq "ActiveState=active"; do sleep 0.5; done
+
+    echo "Then systemctl reports the status of the service as loaded, active and running"
+    expected='(?s).*?Loaded: loaded .*?Active: active \(running\)'
+    systemctl status "$SERVICE_NAME" | grep -Pqz "$expected"
+
+    echo "When the service initialization is completed"
+    retry -n 20 --wait 1 stat /var/snap/test-snapd-success-exit-status/common/ready
+
+    echo "Terminate the service with success exit status 250"
+    systemctl kill -s USR2 --wait "$SERVICE_NAME"
+
+    echo "When the signal is propagated to the service"
+    retry -n 20 --wait 1 sh -c "systemctl show -p ActiveState '$SERVICE_NAME' | MATCH 'ActiveState=inactive'"
+
+    echo "Then systemctl reports the status of the service as terminated"
+    expected='(?s).*?\(code=exited, status=250\)'
+    systemctl status "$SERVICE_NAME" | grep -Pqz "$expected"
+
```

Comment:
and here you could double check that `/var/snap/test-snapd-success-exit-status/common/ready` is gone, indicating that the trap handler executed